Cabal/Distribution/Simple/Build.hs

```diff
@@ -171,8 +172,9 @@ repl pkg_descr lbi flags suffixes args = do
   let clbi = targetCLBI target
       comp = targetComponent target
       lbi' = lbiForComponent comp lbi
+      replFlags = toNubListR $ replReplOptions flags
```

Comment:
Is this going to be sane if, e.g., the repl options are `-trust base -trust containers`? Or is that somehow guaranteed not to happen and the only things in `replReplOptions` are safe to de-dupe?

Reply:
Good question, I'd have to test that a bit. Most of the obvious REPL flags are probably not affected and I'm not sure "-trust" even makes sense here? (Since that seems like it should affect the build outcome?). Sadly there's no real "safe" way of deduping stuff (I'm not even sure the current use of NubListR is completely sensible in the presence of space separated flags?)

A simple solution might be to simply break the NubListR invariant by manually wrapping the list with the newtype, that avoids any doubt about whether things could break (but is rather hacky)...

Reply:
Yeah, I can't say definitively if this is a problem either. It just looks suspicious to me in the context of #5369, which removed a load of `nub` operations on argument lists.

Reply:
Oh, actually, master no longer uses NubListR it seems. I only used that because I was forced by the types of the GHC config. So seems the problem is trivially solved by rebasing onto master and just eliminating NubListR entirely. I wasn't using it myself anyway.